authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/role/RolePolicyProvider.java

```diff
@@ -67,8 +71,16 @@ public void evaluate(Evaluation evaluation) {
                 } else if (hasRole) {
                     evaluation.grant();
                 }
+            } else {
+                iterator.remove();
```

Comment:
The only reason I'm not 100% sure is that we can potentially add more overhead during policy evaluation and send multiple updates to the database if a role is shared by multiple policies.

Perhaps we should just ignore the missing role here and eventually update the configuration when eventually updating the policy. Or we can promptly update the config whenever we return the representation, as you are doing when removing the role from the representation.

wdyt?

Reply:
You're right. Plus, since we are getting the policy's representation (and therefore removing the non-existing roles), we should never satisfy the `role == null` condition during evaluation anyway.

I think I will only update the configuration when updating the policy. This seems like the most efficient way of doing this.